tests/backup/backup_test.go

```diff
@@ -334,6 +332,511 @@ var _ = Describe("{BackupCreateKillStoreRestore}", func() {
+						go func(provider, namespace string) {
+							clusterName := fmt.Sprintf("%s-%s", ClusterName, provider)
+							backupName := fmt.Sprintf("%s-%s-%s", BackupNamePrefix, provider, namespace)
+							orgID := fmt.Sprintf("%s-%s-%s", strings.ToLower(taskNamePrefix),
+								provider, Inst().InstanceID)
+							Step(fmt.Sprintf("Delete backup full name %s:%s:%s",
+								clusterName, namespace, backupName), func() {
+								backupDriver := Inst().Backup
+								bkpDeleteRequest := &api.BackupDeleteRequest{
+									Name:  backupName,
+									OrgId: orgID,
+								}
+								_, err := backupDriver.DeleteBackup(bkpDeleteRequest)
+
+								ctx, _ := context.WithTimeout(context.Background(),
+									BackupRestoreCompletionTimeoutMin*time.Minute)
+
+								logrus.Infof("Wait for backup %s is deleted", backupName)
+								if err := backupDriver.WaitForBackupDeletion(ctx, backupName, orgID,
+									BackupRestoreCompletionTimeoutMin*time.Minute,
+									RetrySeconds*time.Second); err != nil {
+									errChan <- err
+									return
+								}
+
+								errChan <- err
+							})
+						}(provider, namespace)
+
+						go func(provider, namespace string) {
+							clusterName := fmt.Sprintf("%s-%s", ClusterName, provider)
+							restoreName := fmt.Sprintf("%s-%s-%s", RestoreNamePrefix, provider, namespace)
+							orgID := fmt.Sprintf("%s-%s-%s", strings.ToLower(taskNamePrefix),
+								provider, Inst().InstanceID)
+							Step(fmt.Sprintf("Delete restore full name %s:%s:%s",
+								clusterName, namespace, restoreName), func() {
+								backupDriver := Inst().Backup
+								deleteRestoreReq := &api.RestoreDeleteRequest{
+									OrgId: orgID,
+									Name:  restoreName,
+								}
+								_, err := backupDriver.DeleteRestore(deleteRestoreReq)
+
+								ctx, _ := context.WithTimeout(context.Background(),
+									BackupRestoreCompletionTimeoutMin*time.Minute)
+
+								logrus.Infof("Wait for restore %s is deleted", restoreName)
+								if err := backupDriver.WaitForRestoreDeletion(ctx, restoreName, orgID,
+									BackupRestoreCompletionTimeoutMin*time.Minute,
+									RetrySeconds*time.Second); err != nil {
+									errChan <- err
+									return
+								}
+
+								errChan <- err
+							})
+						}(provider, namespace)
+					}
+
+					for i := 0; i < len(providerNamespaces)*2; i++ {
+						select {
+						case <-ctx.Done():
+							Expect(err).NotTo(HaveOccurred(),
+								fmt.Sprintf("Failed to complete backup for provider %s cluster. Error: [%v]", provider, ctx.Err()))
+						case err := <-errChan:
+							Expect(err).NotTo(HaveOccurred(),
+								fmt.Sprintf("Failed to complete backup for provider %s cluster. Error: [%v]", provider, err))
+						}
+					}
+				}
+			})
+		}
+
+		Step("teardown backup objects for test", func() {
+			for _, provider := range providers {
+				providerClusterConfigPath, err := getProviderClusterConfigPath(provider, kubeconfigList)
```

Comment:
Can we use existing `TearDownBackupRestore()` function with additional input param for provider?

Reply:
Well, this will require changes to utility functions we use, because they consume provider from env variable. This will be done in refactoring part.